large-message-core/src/main/java/com/bakdata/kafka/CompressionType.java

```diff
@@ -0,0 +1,175 @@
+        }
+
+        @Override
+        public byte[] decompress(byte[] bytes) {
+            return CompressionType.decompress(org.apache.kafka.common.record.CompressionType.GZIP, bytes);
+        }
+    },
+
+    SNAPPY(2, "snappy") {
+        @Override
+        public byte[] compress(byte[] bytes) {
+            return CompressionType.compress(org.apache.kafka.common.record.CompressionType.SNAPPY, bytes);
+        }
+
+        @Override
+        public byte[] decompress(byte[] bytes) {
+            return CompressionType.decompress(org.apache.kafka.common.record.CompressionType.SNAPPY, bytes);
+        }
+    },
+
+    LZ4(3, "lz4") {
+        @Override
+        public byte[] compress(byte[] bytes) {
+            return CompressionType.compress(org.apache.kafka.common.record.CompressionType.LZ4, bytes);
+        }
+
+        @Override
+        public byte[] decompress(byte[] bytes) {
+            return CompressionType.decompress(org.apache.kafka.common.record.CompressionType.LZ4, bytes);
+        }
+    },
+
+    ZSTD(4, "zstd") {
+        @Override
+        public byte[] compress(byte[] bytes) {
+            return CompressionType.compress(org.apache.kafka.common.record.CompressionType.ZSTD, bytes);
+        }
+
+        @Override
+        public byte[] decompress(byte[] bytes) {
+            return CompressionType.decompress(org.apache.kafka.common.record.CompressionType.ZSTD, bytes);
+        }
+    };
+
+    @Getter
+    private final byte id;
+    @NonNull
+    @Getter
+    private final String name;
+
+    private static byte[] compress(org.apache.kafka.common.record.CompressionType compressionType, byte[] bytes) {
+        final ByteBufferOutputStream outStream = new ByteBufferOutputStream(bytes.length);
+        try (OutputStream stream = compressionType.wrapForOutput(outStream, RecordBatch.MAGIC_VALUE_V2)) {
+            stream.write(bytes);
+            stream.flush();
+        } catch (IOException e) {
+            throw new SerializationException("Failed to compress with type " + compressionType, e);
+        }
+
+        return outStream.buffer().array();
+    }
+
+    private static final BufferSupplier bufferSupplier = BufferSupplier.create();
+
+    CompressionType(int id, String name) {
+        this.id = (byte)id;
+        this.name = name;
+    }
+
+    public static final String HEADER_NAME = HeaderLargeMessagePayloadProtocol.HEADER_PREFIX + ".compression";
+
+    private static byte[] decompress(org.apache.kafka.common.record.CompressionType compressionType, byte[] bytes) {
+        try (InputStream stream = compressionType.wrapForInput(ByteBuffer.wrap(bytes), RecordBatch.MAGIC_VALUE_V2, bufferSupplier)) {
+            return IoUtils.toByteArray(stream);
+        } catch (IOException e) {
+            throw new SerializationException("Failed to compress with type " + compressionType, e);
+        }
+    }
+
+    public static CompressionType forId(int id) {
```

Comment:
Our code style should move this before private methods. This can also be package private